src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -596,19 +598,19 @@ protected async Task<IEnumerable<ContentItem>> BuildNewVersionsAsync(IEnumerable
 
         public async Task CreateAsync(ContentItem contentItem, VersionOptions options)
         {
-            if (string.IsNullOrEmpty(contentItem.ContentItemVersionId))
-            {
-                contentItem.ContentItemVersionId = _idGenerator.GenerateUniqueId(contentItem);
-                contentItem.Published = true;
-                contentItem.Latest = true;
```

Comment:
We don't rely on a ContentItemVersionId not set on the ContentItem to define if Latest or not.

Reply:
Have you tried the cloning events after this change? Or the code path in drivers that check for `IsNew` ? I have a weird feeling that is was there for a reason.

Reply:
Will try.

Reply:
Other example:

```c#
        public async Task<ContentValidateResult> RestoreAsync(ContentItem contentItem)
        {
            // Prepare record for restore.
            // So that a new record will be created.
            contentItem.Id = 0;
            // So that a new version id will be generated.
            contentItem.ContentItemVersionId = "";
            contentItem.Latest = contentItem.Published = false;

```

Reply:
Yeah I had the same thought. Remembered that all we do is set the ContentItemVersionId to String.Empty is some use cases which should be fine.

Reply:
Note to myself. Investigate when this ContentItemVersionId conditional check has been added and why.

Reply:
As far as history goes: https://github.com/OrchardCMS/OrchardCore/pull/749/files#diff-a5ffffe50b38039080ea22d032ad1f66322d17f108ff9233bcc3a77f59d9cf03L341

Reply:
Cloning worked else the unit tests would have failed.

Reply:
Restore seems to have issues.

Reply:
Restore fixed

Reply:
For CloningAsync and ClonedAsync the only difference is that now the ContentItemVersionId will be different than null. Should not be affecting our current implementations.

Reply:
The only thing I'm not sure about is the IsNew code path in drivers. Right now I'm assuming that some unit test would have already failed.